src/odbc/odbc.c

```diff
@@ -1453,54 +1513,105 @@ _SQLBindParameter(SQLHSTMT hstmt, SQLUSMALLINT ipar, SQLSMALLINT fParamType, SQL
 		ODBC_EXIT_(stmt);
 	}
 
-	/* fill APD related fields */
-	apd = stmt->apd;
-	orig_apd_size = apd->header.sql_desc_count;
-	if (ipar > apd->header.sql_desc_count && desc_alloc_records(apd, ipar) != SQL_SUCCESS) {
-		odbc_errs_add(&stmt->errs, "HY001", NULL);
-		ODBC_EXIT_(stmt);
-	}
-	drec = &apd->records[ipar - 1];
+	/* If the parameter focus is set to 0, bind values as arguments */
+	/* Otherwise, bind values as the columns of a table-valued parameter */
+	if (stmt->param_focus == 0) {
+		/* fill APD related fields */
+		apd = stmt->apd;
+		orig_apd_size = apd->header.sql_desc_count;
+		if (ipar > apd->header.sql_desc_count && desc_alloc_records(apd, ipar) != SQL_SUCCESS) {
+			odbc_errs_add(&stmt->errs, "HY001", NULL);
+			ODBC_EXIT_(stmt);
+		}
+		drec = &apd->records[ipar - 1];
 
-	if (odbc_set_concise_c_type(fCType, drec, 0) != SQL_SUCCESS) {
-		desc_alloc_records(apd, orig_apd_size);
-		odbc_errs_add(&stmt->errs, "HY004", NULL);
-		ODBC_EXIT_(stmt);
-	}
+		if (odbc_set_concise_c_type(fCType, drec, 0) != SQL_SUCCESS) {
+			desc_alloc_records(apd, orig_apd_size);
+			odbc_errs_add(&stmt->errs, "HY004", NULL);
+			ODBC_EXIT_(stmt);
+		}
 
-	stmt->need_reprepare = 1;
+		stmt->need_reprepare = 1;
 
-	/* TODO other types ?? handle SQL_C_DEFAULT */
-	if (drec->sql_desc_type == SQL_C_CHAR || drec->sql_desc_type == SQL_C_WCHAR || drec->sql_desc_type == SQL_C_BINARY)
-		drec->sql_desc_octet_length = cbValueMax;
-	drec->sql_desc_indicator_ptr = pcbValue;
-	drec->sql_desc_octet_length_ptr = pcbValue;
-	drec->sql_desc_data_ptr = (char *) rgbValue;
+		/* TODO other types ?? handle SQL_C_DEFAULT */
+		if (drec->sql_desc_type == SQL_C_CHAR || drec->sql_desc_type == SQL_C_WCHAR || drec->sql_desc_type == SQL_C_BINARY)
+			drec->sql_desc_octet_length = cbValueMax;
+		drec->sql_desc_indicator_ptr = pcbValue;
+		drec->sql_desc_octet_length_ptr = pcbValue;
+		drec->sql_desc_data_ptr = (char *) rgbValue;
 
-	/* field IPD related fields */
-	ipd = stmt->ipd;
-	orig_ipd_size = ipd->header.sql_desc_count;
-	if (ipar > ipd->header.sql_desc_count && desc_alloc_records(ipd, ipar) != SQL_SUCCESS) {
-		desc_alloc_records(apd, orig_apd_size);
-		odbc_errs_add(&stmt->errs, "HY001", NULL);
-		ODBC_EXIT_(stmt);
-	}
-	drec = &ipd->records[ipar - 1];
+		if (fSqlType == SQL_SS_TABLE) {
```

Comment:
If we specify `ParameterType=SQL_SS_TABLE`, allocate a `SQLTVP` buffer to store table-related information. The type  of the TVP is passed as a `char*` buffer in `rgbValue`, and should have the format `schema.typename`, or just `typename`.

The table buffer is stored in `stmt->apd->records[index - 1].sql_desc_data_ptr`, where `stmt->apd` stores information about outgoing data.